### PR TITLE
Warn on requiring deprecated files

### PR DESCRIPTION
**Pull request by maclover7**
# Description:

Both of these files are deprecated, so we should warn users if they are requiring them, since they will be removed in a future version of Rubygems.

---
# Tasks:
- [ ] Describe the problem / feature
- [ ] Write tests
- [ ] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by segiddins** [comment]
These need to respect `Gem::Deprecate.skip`

**Comment by segiddins** [comment]
Ping @maclover7 ?

File: lib/rubygems/source_local.rb [hunk 1]
```diff
@@ -2,5 +2,4 @@
 require 'rubygems/source'
 require 'rubygems/source_local'
 
-# TODO warn upon require, this file is deprecated.
-
+Kernel.warn "Requiring rubygems/source_local is deprecated; please use rubygems/source/local instead."
```
Comment (duckinator) [inline lib/rubygems/source_local.rb]: @segiddins I attempted to make a new version of this PR that uses `Gem.location_of_caller`, but...

in source_local.rb:

```
  Kernel.warn "#{Gem.location_of_caller.join(':')}: Warning: Requiring rubygems/source_local is deprecated; please use rubygems/source/local instead."
```

x.rb:

```ruby
require 'rubygems/source_local'
```

`ruby -Ilib x.rb` output:

```
/home/ellen/dev/ruby/rubygems/rubygems/lib/rubygems/core_ext/kernel_require.rb:54: Warning: Requiring rubygems/source_local is deprecated; please use rubygems/source/local instead.
```

am I missing something?

**Comment by segiddins** [comment]
@duckinator it might need to be the caller's caller's location then

**Comment by duckinator** [comment]
apparently it needs the caller's caller's caller's location, and I hate computers.

**Comment by duckinator** [comment]
Closing in favor of #1939.

**Pull request closed without merging**